fdbcli/fdbcli.actor.cpp

```diff
@@ -406,6 +406,28 @@ static std::vector<std::vector<StringRef>> parseLine(std::string& line, bool& er
 			case ';':
 				line.erase(i, 1);
 				break;
+			// Handle \uNNNN utf-8 characters from JSON strings but only as a single byte
+			// Return an error for a sequence out of range for a single byte
+			case 'u': {
+				if (i + 6 > line.length()) {
+					err = true;
+					ret.push_back(std::move(buf));
+					return ret;
+				}
+				char* pEnd;
+				save = line[i + 6];
+				line[i + 6] = 0;
+				unsigned long val = strtoul(line.data() + i + 2, &pEnd, 16);
+				ent = char(val);
```

Comment:
Does this work for all the possibilities of a unicode escape? Do we need it to?

Reply:
It doesn't, and we don't need it as far as I know.  I'm also not sure how to support it, so I chose to just error.  I didn't see a way to return a descriptive error from this function.

I added this because we do print keys in JSON form in some places in FDB, including the `rangeconfig show` output but for all cases in FDB where keys are serialized as JSON strings the nonprintable characters are single-byte unicode sequences.  For example

```
fdb> rangeconfig set \x01\x02\x03\x04 \xff replication 5
...
         "begin" : "\u0001\u0002\u0003\u0004",
...
            "end" : "\u00FF"
```